source/sigv4.c

```diff
@@ -351,45 +351,61 @@ static void setQueryParameterValue( size_t currentParameter,
                                     CanonicalContext_t * pCanonicalRequest );
 
 /**
- * @brief Update the HMAC using an input key.
+ * @brief Generates the key for the HMAC operation.
+ *
  * @note This function can be called multiple times before calling
- * #hmacData. Appending multiple substrings, then calling #hmacKey
- * on the appended string is also equivalent to calling #hmacKey on
+ * #hmacIntermediate. Appending multiple substrings, then calling #hmacAddKey
+ * on the appended string is also equivalent to calling #hmacAddKey on
  * each individual substring.
  * @note This function accepts a const char * so that string literals
  * can be passed in.
  *
  * @param[in] pHmacContext The context used for HMAC calculation.
  * @param[in] pKey The key used as input for HMAC calculation.
  * @param[in] keyLen The length of @p pKey.
+ * @param[in] isKeyPrefix Flag to indicate whether the passed key is
+ * prefix of a complete key for an HMAC operation. If this is a prefix,
+ * then it will be stored in cache for use with remaining part of the
+ * key that will be provided in a subsequent call to @ref hmacAddKey.
+ *
  * @return Zero on success, all other return values are failures.
  */
-static int32_t hmacKey( HmacContext_t * pHmacContext,
-                        const char * pKey,
-                        size_t keyLen );
+static int32_t hmacAddKey( HmacContext_t * pHmacContext,
+                           const char * pKey,
+                           size_t keyLen,
+                           bool isKeyPrefix );
```

Comment:
Why bother with another input parameter?

Reply:
Because we want to add the zero padding only when the key is completely received. I thought it is better to encapsulate the `K'` operation within this function than distribute it across two functions